src/Test.Utilities.Sarif/HttpMockHelper.cs

```diff
@@ -116,6 +116,13 @@ private static bool CompareHeaders(HttpRequestHeaders headers1, HttpRequestHeade
 
         protected override Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
         {
+            // In most cases, we set up one request and response pair.
+            // The matching mechanism is only used for those situations where we need more than one pair.
+            if (this.mockedResponses.Count == 1 && mockedResponses[0].Item1.RequestUri != null)
```

Comment:
>mockedResponses[0].Item1.RequestUri != null [](http://example.com/codeflow?start=51&length=43)

1. Please check the pr description for the reason of this change.
2. This also fast return on the case of that we expect to have unhandled exception.
3.  All SPM tests still pass with this change.
4. If this is merged, I can work on change the tests to removed the need of using same timestamps
 e.g. from:
`signer.SignRequest(defaultRequest, "region1", "storage1", timestamp);`
to 
`signer.SignRequest(defaultRequest, "region1", "storage1");`
or just keep them? they works, change or not.
5. the earlier i was thinking about hard code a list of time related strings to skip when compare header, that was discarded: [https://github.com/microsoft/sarif-sdk/pull/2714](https://github.com/microsoft/sarif-sdk/pull/2714)